### ISSUE TITLE
REQ: Option to turn off all HA stuff

### ISSUE THREAD

**Issue by mongojarle** [tesla-local-control/tesla_ble_mqtt_core#17]
I very much like this project, but I'm not using HA at all, just MQTT.
Maybe an ENV (true/false) that could disable everything related to HA.

**Comment by baylanger** [comment]
I didn't expect this one but we can do this ~ likely within next 2 weeks or less.

Comments... and yours are welcome.

- Aside MQTT topics directly related to HA, others will continue to be published (presence, etc).
- The functionality to run or not vehicle's presence detection is still available.
- Support for multi-cars (coming soon) would also work.

## Putting some notes:
1) New feature for Docker standalone product, on/off using ~ HA_BACKEND_DISABLE (true|false);
2) Keys set; Currently generated by a "Push Button" in HA. If HA is disable and at start there's no key available for the VIN, call a function to generate the keys;
3) Instructions; Add a section on this topic.

If needed, will edit the above.

**Comment by baylanger** [comment]
Updated my comment above a few times.

Started to work in this [PR](https://github.com/tesla-local-control/tesla_ble_mqtt_core/pull/18).

If you use the MQTT to publish the generation of pub keys instead of manually, then this PR is most likely ready.

@iainbullock @raphmur anything you think might be missing?

**Comment by raphmur** [comment]
Hi there,
Yes interesting feature.
Indeed, the mqtt topics for the standalone version are homeassistant. An env var is relevant, but then we also need to declare alternative mqtt topics.

**Comment by baylanger** [comment]
> Hi there, Yes interesting feature. Indeed, the mqtt topics for the standalone version are homeassistant. An env var is relevant, but then we also need to declare alternative mqtt topics.
>
🤔  Are you positive the topics are within homeassistant?

- listen_mqtt.sh:
```
eval $MOSQUITTO_SUB_BASE --nodelay -t tesla_ble/+/+
```

discovery.sh:
```
 TOPIC_ROOT=tesla_ble/${vin}
{SNIP}
 eval $MOSQUITTO_PUB_BASE -t homeassistant/binary_sensor/${DEV_ID}/presence/config -m \
  '{
   "state_topic": "'${TOPIC_ROOT}'/binary_sensor/presence",
```

Isn't the HA topic just so the HA understand what each `tesla_ble/...` are all about?

**Comment by iainbullock** [comment]
I asusme the OP wants to control via MQTT but not using HA. Maybe node-red or something else. 

Everything in discovery.sh is then not required to be run as this is what autocreates the device / entities in HA. 

The OP can then interact with the code by sending messages to the relevent topics in MQTT by whatever means. 

This includes creating / deploying the keys

I hope this helps?

**Comment by mongojarle** [comment]
Hi, sorry for my absence!

Yes, @iainbullock is correct. I'm using NodeRed, and only NodeRed. I created the keys by sending the payload "generate_keys" to the MQTT topic "tesla_ble/config". This generated the keys as expected.
I running this as a docker container on a rPi zero 2W, meant to sit in the garage,

I really appreciate all the attention to this, and expedited response!

**Comment by iainbullock** [comment]
@baylanger we don't need to change the generate / deploy keys to a function. The OP has already done this by sending the relevant message to MQTT from node-red

**Comment by raphmur** [comment]
PR #18 merged.
@mongojarle expect update in next release of the [docker version](https://github.com/tesla-local-control/tesla_ble_mqtt_docker).
If you want to anticipate, you can directly pull the code from this repo, without guarantees...

**Comment by baylanger** [comment]
Lets not forget this was merged "accidentally" to multi-cars branch.

@mongojarle if you know your ways around, use branch multi-cars otherwise hold on ~ Friday this should be merged.

**Comment by raphmur** [comment]
Right
Reopening to keep track until merged into main

**Comment by raphmur** [comment]
Merged into v0.1.0 releases.
Docker: https://github.com/tesla-local-control/tesla_ble_mqtt_docker/releases/tag/v0.1.0
HA Addon: https://github.com/tesla-local-control/tesla-local-control-addon/releases/tag/v0.1.0

Please update and tell if this is ok for you. You will need to edit this in [`stack.env`](https://github.com/tesla-local-control/tesla_ble_mqtt_docker/blob/main/stack.env):

```# WARNING; If you run Home Assistant, keep this true unless you know what you're doing
#
ENABLE_HA_FEATURES=true```
